src/transactions/transactionBuilder/remoteAbi.ts

```diff
@@ -262,89 +261,91 @@ function parseArg(
  * @param arg
  * @param position
  */
-function checkType(param: TypeTag, arg: EntryFunctionArgumentTypes, position: number) {
+function checkType(param: TypeTag, arg: EntryFunctionArgumentTypes, position: number): EntryFunctionArgumentTypes {
   if (param.isBool()) {
     if (isBcsBool(arg)) {
-      return;
+      return arg;
     }
     throwTypeMismatch("Bool", position);
   }
   if (param.isAddress()) {
     if (isBcsAddress(arg)) {
-      return;
+      return arg;
     }
     throwTypeMismatch("AccountAddress", position);
   }
   if (param.isU8()) {
     if (isBcsU8(arg)) {
-      return;
+      return arg;
     }
     throwTypeMismatch("U8", position);
   }
   if (param.isU16()) {
     if (isBcsU16(arg)) {
-      return;
+      return arg;
     }
     throwTypeMismatch("U16", position);
   }
   if (param.isU32()) {
     if (isBcsU32(arg)) {
-      return;
+      return arg;
     }
     throwTypeMismatch("U32", position);
   }
   if (param.isU64()) {
     if (isBcsU64(arg)) {
-      return;
+      return arg;
     }
     throwTypeMismatch("U64", position);
   }
   if (param.isU128()) {
     if (isBcsU128(arg)) {
-      return;
+      return arg;
     }
     throwTypeMismatch("U128", position);
   }
   if (param.isU256()) {
     if (isBcsU256(arg)) {
-      return;
+      return arg;
     }
     throwTypeMismatch("U256", position);
   }
   if (param.isVector()) {
     if (arg instanceof MoveVector) {
       // If there's anything in it, check that the inner types match
-      // Note that since it's typed, the first item should be the same as the rest
-      if (arg.values.length > 0) {
-        checkType(param.value, arg.values[0], position);
-      }
+      const newValues = arg.values.map((item) => checkType(param.value, item, position));
+      return new MoveVector(newValues);
+    }
 
-      return;
+    // Special case string as vector<u8>
+    if (arg instanceof MoveString && param.value.isU8()) {
```

Comment:
it still doesnt solve the user simple input `["hello"]`. I think the problem is that we try to convert it into Hex if the type is `vector<u8>` and the arg is `string`. We might want to try do it, and if it doesnt succeed just parse it as a string
https://github.com/aptos-labs/aptos-ts-sdk/blob/main/src/transactions/transactionBuilder/remoteAbi.ts#L207

Reply:
That's super tricky in itself.  What if you want to put the string `0x12345`, and not the bytes `0x12345`?

This variable behavior can be really confusing for a user, and I'd prefer not to have something that chooses this behavior arbitrarily

Similarly, if you try to pass in an address string, say the address "0x1", and you want to store it as a string, and not as 1 byte

Reply:
I feel we mostly expect the users the understand what is right or wrong here. There are some components involved like the right way for a move function to ask for argument types (instead of `vector<vector<u8>>` use `vector<vector<String>>` if expected a string) unless you are intently want to accept  any `u8` value and expect any type to be serialized before).